backend/layers/processing/rollback.py

```diff
@@ -0,0 +1,301 @@
+from backend.layers.thirdparty.uri_provider import UriProvider
+
+logger = logging.getLogger(__name__)
+
+
+class RollbackType(Enum):
+    PRIVATE_COLLECTIONS = "private_collections"
+    PUBLIC_COLLECTIONS = "public_collections"
+    PUBLIC_COLLECTION_LIST = "public_collection_list"
+    PRIVATE_COLLECTION_LIST = "private_collection_list"
+    PRIVATE_DATASET_LIST = "private_dataset_list"
+
+
+class RollbackEntity:
+    def __init__(
+        self, business_logic: BusinessLogic, rollback_type: RollbackType, entity_id_list: List[str] = None
+    ) -> None:
+        self.business_logic = business_logic
+        self.rollback_type = rollback_type
+        self.entity_id_list = entity_id_list
+
+    def rollback(self):
+        if self.rollback_type == RollbackType.PRIVATE_COLLECTIONS:
+            self.rollback_private_collections()
+        elif self.rollback_type == RollbackType.PUBLIC_COLLECTIONS:
+            self.rollback_public_collections()
+        elif self.rollback_type == RollbackType.PUBLIC_COLLECTION_LIST:
+            self.rollback_public_collection_list()
+        elif self.rollback_type == RollbackType.PRIVATE_COLLECTION_LIST:
+            self.rollback_private_collection_list()
+        elif self.rollback_type == RollbackType.PRIVATE_DATASET_LIST:
+            self.rollback_private_dataset_list()
+        else:
+            raise ValueError(f"Invalid rollback type: {self.rollback_type}")
+
+    def rollback_private_collections(self) -> None:
+        """
+        Rollback all the datasets in all private collections. This will restore the
+        state of private collections to their pre-migration state. Then, triggers deletion of the DB
+        references and S3 assets for the rolled back DatasetVersions.
+        """
+        filter = CollectionQueryFilter(is_published=False)
+        collections = self.business_logic.get_collections(filter)
+        rolled_back_datasets = []
+        for collection in collections:
+            collection_datasets = self.rollback_private_collection(collection.version_id)
+            rolled_back_datasets.extend(collection_datasets)
+        self._clean_up_rolled_back_datasets(rolled_back_datasets)
+
+    def rollback_private_collection_list(self) -> None:
+        """
+        Rollback all the datasets from the input list of private collections. This will restore the
+        state of the list of private collections to their pre-migration state. Then, triggers deletion of the DB
+        references and S3 assets for the rolled back DatasetVersions.
+        """
+        rolled_back_datasets = []
+        for collection_version_id in self.entity_id_list:
+            try:
+                collection_dataset_versions = self.rollback_private_collection(
+                    CollectionVersionId(collection_version_id)
+                )
+            except Exception as e:
+                logger.info(f"Failed to rollback CollectionVersion {collection_version_id}, {e}")
```

Comment:
- change `info` to `exception` and remove `{e}`. 
- add that it was a private collection to error message